### ISSUE TITLE
Regression: OSError: [Errno 42] Protocol not available

### ISSUE THREAD

**Issue by tothegump** [celery/py-amqp#196]
# Platform
* macOS High Sierra
* Version 10.13.4
* MacBook Pro (Retina, 13-inch, Early 2015)
* Python 3.5.2

add a celery report:
```
software -> celery:4.2.0rc4 (windowlicker) kombu:4.2.0 py:3.5.2
            billiard:3.5.0.3 py-amqp:2.3.0
platform -> system:Darwin arch:64bit imp:CPython
loader   -> celery.loaders.default.Loader
settings -> transport:amqp results:disabled
```
# Reproduce
Run this code snippet:
```Python
from amqp import Connection
conn = Connection()
conn.connect()
````

from commit `73391a7` to `3973415` (current master HEAD) raise errors:
```
Traceback (most recent call last):
  File "/Users/xxx/.virtualenvs/a_env/lib/python3.5/site-packages/IPython/core/interactiveshell.py", line 2910, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-2-e6621bf59778>", line 3, in <module>
    conn.connect()
  File "/Users/xxx/s_dir/py-amqp/amqp/connection.py", line 279, in connect
    self.transport.connect()
  File "/Users/xxx/s_dir/py-amqp/amqp/transport.py", line 81, in connect
    self.socket_settings, self.read_timeout, self.write_timeout,
  File "/Users/xxx/s_dir/py-amqp/amqp/transport.py", line 165, in _init_socket
    self._set_socket_options(socket_settings)
  File "/Users/xxx/s_dir/py-amqp/amqp/transport.py", line 208, in _set_socket_options
    self.sock.setsockopt(SOL_TCP, opt, val)
OSError: [Errno 42] Protocol not available
```
when I checkout to commits from `3faa759` to `6a6127c` (the commit before `73391a7`) and run that code, rase error:
```
Traceback (most recent call last):
  File "/Users/xxx/.virtualenvs/a_env/lib/python3.5/site-packages/IPython/core/interactiveshell.py", line 2910, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-2-e6621bf59778>", line 3, in <module>
    conn.connect()
  File "/Users/xxx/s_dir/py-amqp/amqp/connection.py", line 282, in connect
    self.transport.connect()
  File "/Users/xxx/s_dir/py-amqp/amqp/transport.py", line 85, in connect
    self.socket_settings, self.read_timeout, self.write_timeout,
  File "/Users/xxx/s_dir/py-amqp/amqp/transport.py", line 137, in _init_socket
    self._set_socket_options(socket_settings)
  File "/Users/xxx/s_dir/py-amqp/amqp/transport.py", line 176, in _set_socket_options
    tcp_opts = self._get_tcp_socket_defaults(self.sock)
  File "/Users/xxx/s_dir/py-amqp/amqp/transport.py", line 172, in _get_tcp_socket_defaults
    tcp_opts[enum] = sock.getsockopt(SOL_TCP, opt)
TypeError: an integer is required (got type str)
```

when I checkout to commit `fbe27f1` (the commit before `3faa759`), and run that code, nothing wrong.

**Comment by clokep** [comment]
This was fixed in 2.3.2, I believe. See #194

**Comment by tothegump** [comment]
@clokep 
Yes, it is.
My fault. I did not get the newest commit.
Thanks.
